repair_shards.c

```diff
@@ -209,3 +276,99 @@ RecreateTableDDLCommandList(Oid relationId, int64 shardId)
 
 	return extendedRecreateCommandList;
 }
+
+
+/*
+ * CopyDataFromFinalizedPlacement copies a the data for a shard (identified by
+ * a relation and shard identifier) from a healthy placement to one needing
+ * repair. The unhealthy placement must already have an empty relation in place
+ * to receive rows from the healthy placement. This function returns a boolean
+ * indicating success or failure.
+ */
+static bool
+CopyDataFromFinalizedPlacement(Oid distributedTableId, int64 shardId,
+                               ShardPlacement *healthyPlacement,
+                               ShardPlacement *placementToRepair)
+{
+	char *relationName = get_rel_name(distributedTableId);
+	char relationKind = get_rel_relkind(distributedTableId);
+	const char *shardName = NULL;
+	char *nodeName = placementToRepair->nodeName;
+	int32 nodePort = placementToRepair->nodePort;
+	StringInfo copyRelationQuery = makeStringInfo();
+
+	PGconn *connection = NULL;
+	PGresult *result = NULL;
+
+	if (relationKind == RELKIND_FOREIGN_TABLE)
```

Comment:
Pretty minor. What if we moved /grouped the ``char relationKind = get_rel_relkind(..)`` declaration with this ``if`` block?